common/ratelimit/limiter.go

```diff
@@ -113,13 +115,11 @@ func (d *rateLimiter) checkAllowed(ctx context.Context, params common.RequestPar
 		// Determine bucket deduction
 		deduction := time.Microsecond * time.Duration(1e6*float32(params.BlobSize)/float32(params.Rate)/d.globalRateParams.Multipliers[i])
 
-		prevLevel := bucketParams.BucketLevels[i]
-
 		// Update the bucket level
-		bucketParams.BucketLevels[i] = getBucketLevel(bucketParams.BucketLevels[i], size, interval, deduction)
-		allowed = allowed && bucketParams.BucketLevels[i] > 0
+		bucketLevels[i] = getBucketLevel(bucketParams.BucketLevels[i], size, interval, deduction)
+		allowed = allowed && bucketLevels[i] > 0
 
-		d.logger.Debug("Bucket level updated", "key", params.RequesterID, "name", params.RequesterName, "prevLevel", prevLevel, "level", bucketParams.BucketLevels[i], "size", size, "interval", interval, "deduction", deduction, "allowed", allowed)
+		d.logger.Debug("Bucket level updated", "key", params.RequesterID, "name", params.RequesterName, "prevLevel", bucketParams.BucketLevels[i], "level", bucketLevels[i], "size", size, "interval", interval, "deduction", deduction, "allowed", allowed)
 
 		// Update metrics only if the requester name is provided. We're making
```

Comment:
Should we update the metric to use  `bucketLevels[i]`?

Reply:
Updated